services/src/main/java/org/keycloak/services/resources/admin/AdminRoot.java

```diff
@@ -97,6 +99,11 @@ public static UriBuilder adminBaseUrl(UriBuilder base) {
      */
     @GET
     public Response masterRealmAdminConsoleRedirect() {
+
+        if (!(Profile.isFeatureEnabled(Profile.Feature.ADMIN2) || Profile.isFeatureEnabled(Profile.Feature.ADMIN))) {
+            return Response.noContent().build();
```

Comment:
Should this be not-found rather than an empty response?

Reply:
Yep that makes more sense, I'll adapt the code.